### PR TITLE
Remove `voiceSearchHelper` from `AppDependencyProvider`

### PR DESCRIPTION
**Pull request by samsymons**
Task/Issue URL: https://app.asana.com/0/414235014887631/1208574561964832/f
Tech Design URL:
CC:

**Description**:

This PR removes `voiceSearchHelper` from `AppDependencyProvider` and updates the test suite to no longer mutate the dependency provider singleton. Tests that use this pattern have been prone to occasional crashes, so I'm simplifying them and working towards removing `makeTestingInstance()` entirely.

**Steps to test this PR**:
1. Check that voice search works as expected

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/SmallOmniBarState.swift [hunk 1]
```diff
@@ -30,25 +30,27 @@ struct SmallOmniBarState {
         let showShareButton: Bool = false
         let clearTextOnStart = true
         let allowsTrackersAnimation = false
-        let showSearchLoupe = !AppDependencyProvider.shared.voiceSearchHelper.isSpeechRecognizerAvailable
```
Comment (samsymons) [inline DuckDuckGo/SmallOmniBarState.swift:33]: I notice that sometimes this value uses `isSpeechRecognizerAvailable`, and in other places it uses `isVoiceSearchEnabled`. Intentional?

Comment (Bunn) [reply DuckDuckGo/SmallOmniBarState.swift:33]: Yes, the naming could be better but `isSpeechRecognizerAvailable` is about having the speechRecognizer feature on iOS itself available whereas `isVoiceSearchEnabled` is our feature, if the user turned it on/off.
i.e: `isVoiceSearchEnabled` will always be false if `isSpeechRecognizerAvailable` is false.

In some places (iirc) we have different behaviors depending on system settings or feature settings.

Maybe a better name would be:
* isSystemSpeechRecognizerAvailable
* isSFSpeechRecognizerAvailable

Feel free to change it.

File: DuckDuckGo/SmallOmniBarState.swift [hunk 2]
```diff
@@ -204,24 +216,26 @@ struct SmallOmniBarState {
...
         var onBrowsingStartedState: OmniBarState { return self }
-        var onBrowsingStoppedState: OmniBarState { return HomeEmptyEditingState() }
-        var onEnterPadState: OmniBarState { return LargeOmniBarState.BrowsingTextEditingState() }
+        var onBrowsingStoppedState: OmniBarState { return HomeEmptyEditingState(voiceSearchHelper: voiceSearchHelper) }
+        var onEnterPadState: OmniBarState { return LargeOmniBarState.BrowsingTextEditingState(voiceSearchHelper: voiceSearchHelper) }
         var onEnterPhoneState: OmniBarState { return self }
-        var onReloadState: OmniBarState { return BrowsingTextEditingStartedState() }
+        var onReloadState: OmniBarState { return BrowsingTextEditingStartedState(voiceSearchHelper: voiceSearchHelper) }
+        var showSearchLoupe: Bool { !voiceSearchHelper.isVoiceSearchEnabled }
+        var showVoiceSearch: Bool { !voiceSearchHelper.isVoiceSearchEnabled }
```
Comment (Bunn) [inline DuckDuckGo/SmallOmniBarState.swift]: This is wrong, it should be without the `!`
ex: `var showVoiceSearch: Bool { voiceSearchHelper.isVoiceSearchEnabled }`

**Review by Bunn — CHANGES_REQUESTED** [review]
Overall LGTM, but there's one breaking change there which is why I'm marking as request changes.

File: DuckDuckGo/SmallOmniBarState.swift (hunk 2 above)
Comment (samsymons) [reply DuckDuckGo/SmallOmniBarState.swift]: Good catch, fixed

File: DuckDuckGo/SmallOmniBarState.swift (hunk 1 above)
Comment (samsymons) [reply DuckDuckGo/SmallOmniBarState.swift:33]: That's okay, I'm happy to keep the naming to be honest, just wanted to make sure this was intentional. :+1:

**Review by Bunn — APPROVED** [review]

**Pull request merged**